src/components/recipes/delete/RecipesDelete.test.tsx

```diff
@@ -0,0 +1,59 @@
+import { UseMutationResult } from '@tanstack/react-query';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+import { BrowserRouter } from 'react-router-dom';
+import React from 'react';
+import { useDeleteRecipe } from 'hooks/recipes';
+import RecipesDelete from './RecipesDelete';
+import Factory from 'tests/Factory';
+
+jest.mock('hooks/recipes', () => ({
+    useDeleteRecipe: jest.fn(),
+}));
+const mockNavigate = jest.fn();
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    useNavigate: () => mockNavigate,
+}));
+const mockedUseDeleteRecipe = useDeleteRecipe as jest.MockedFunction<typeof useDeleteRecipe>;
+
+describe('RecipesDelete Component', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        mockedUseDeleteRecipe.mockReturnValue({
+            mutate: jest.fn((recipeId, options) => {
+                if (options?.onSuccess) {
+                    options.onSuccess();
+                }
+            }),
+        } as unknown as UseMutationResult<void, Error, string, unknown>);
```

Comment:
Took a while to resolve this type error, not sure if this is the best way still.

Reply:
I feel like there's a _different_ way (not necessarily better) to do this, let me have a look tomorrow and get back to you...

Reply:
Was just checking how I did this, you can mock the `useDeleteRecipe` hook as below with just a simple `jest.fn()`, then you won't get the type issues:

`const mockedUseDeleteRecipe = jest.fn();`

_But_, you also won't get any type checking when mocking the return value, so it's definitely not a straight improvement.

Reply:
I'll give it a go and see if this works fine. Thanks Matt 👍